packages/mui-material/src/AvatarGroup/AvatarGroup.js

```diff
@@ -108,10 +108,17 @@ const AvatarGroup = React.forwardRef(function AvatarGroup(inProps, ref) {
   const extraAvatars = Math.max(totalAvatars - clampedMax, totalAvatars - maxAvatars, 0);
   const extraAvatarsElement = renderSurplus ? renderSurplus(extraAvatars) : `+${extraAvatars}`;
 
-  const marginValue =
-    ownerState.spacing && SPACINGS[ownerState.spacing] !== undefined
-      ? SPACINGS[ownerState.spacing]
-      : -ownerState.spacing || -8;
+  let marginValue;
+
+  if (ownerState.spacing && SPACINGS[ownerState.spacing] !== undefined) {
+    marginValue = SPACINGS[ownerState.spacing];
+  } else if (ownerState.spacing === 0) {
+    marginValue = 0;
+  } else if (ownerState.spacing === Infinity || ownerState.spacing === -Infinity) {
+    marginValue = SPACINGS.medium;
```

Comment:
We shouldn't handle this. It's the developer's responsibility not to pass `Infinity` or `NaN` to the `spacing`  prop.
```suggestion
```